source/server/session_resource_repository.h

```diff
@@ -229,8 +237,8 @@ TResourceHandle SessionResourceRepository<TResourceHandle>::ResourceMap::remove_
 {
   MapLock lock(map_mutex_);
   auto handle = get_handle(session_id);
+  map_.erase(session_id);
```

Comment:
Not sure I understand this change.

Reply:
Yeah, I had that confused. I thought that `get_handle` used the reverse map. And we needed to erase the session even if that mapping wasn't present. But `get_handle` uses the "forward map" so it should still always be there. Anyway it's not impacted by this change.

Also a good reminder I should add a test for removing dependent sessions to check this code. Or at least double check that it'd getting covered.